### PR TITLE
Persist partition

### LINKED ISSUE #950: kafka-cluster partitions don't necessarily match kafka-mdm-in partitions
Data comes into Metrictank on the data topic (let's call that `mdm`) and when chunks are persisted to cassandra, summaries of the chunk are sent to the persist topic (let's call that `persist`). On start up, MT uses `persist` to avoid overwriting chunks in cassandra that were already persisted. This means that it is important that the summaries in `persist` line up with the data in `mdm` for the instance handling a given partition.

1. MT doesn't really enforce *how* the data in `mdm` is partitioned, just that partitioning is consistent.
2. Partitioning in `persist` is either "ByOrg" or "BySeries" (see [here](https://github.com/grafana/metrictank/blob/85a628a3771b2961d324e1eed2ef01df3bf2e3bf/mdata/notifierKafka/notifierKafka.go#L257) and [here](https://github.com/grafana/metrictank/blob/d12b42fc10dec9486b8184bd5ff7efd39aa48aab/metrictank-sample.ini#L296)). If this isn't how data in `mdm` is partitioned, there is trouble.

It seems to me that the simpler solution is to simply use def.Partition to put the summaries into `persist`.

I am rolling out this change on our side and can submit a cleanup PR. I'm not sure if removing the parameter entirely is backwards compatible or not, however.

### PR DESCRIPTION
**Pull request by shanson7**
Fixes  #950

### CODE REVIEW

File: mdata/notifierKafka/notifierKafka.go [hunk 1]
```diff
@@ -253,15 +253,10 @@ func (c *NotifierKafka) flush() {
...
-		if err != nil {
-			log.Fatal(4, "Unable to get partitionKey for metricDef with id %s. %s", def.Id, err)
-		}
 		kafkaMsg := &sarama.ProducerMessage{
-			Topic: topic,
-			Value: sarama.ByteEncoder(buf.Bytes()),
-			Key:   sarama.ByteEncoder(key),
+			Topic:     topic,
+			Value:     sarama.ByteEncoder(buf.Bytes()),
+			Partition: def.Partition,
```
Comment (Dieterbe) [inline mdata/notifierKafka/notifierKafka.go]: :heart_eyes:

**Comment by replay** [comment]
This one could probably be removed as well: https://github.com/bloomberg/metrictank/blob/26003a6ab728b7206539d764aa4a8e3fe469ccf8/docker/docker-cluster/metrictank.ini#L294

**Comment by replay** [comment]
When upgrade from pre-this-commit to this commit, there should be no issues, right? Because the partition that gets assigned to the persist messages of a topic should remain the same as before

**Comment by shanson7** [comment]
Right. The only case where it should change is if it was doing the wrong thing before.

**Pull request merged**